src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ConstructorSymbolReferenceFinder.cs

```diff
@@ -208,16 +219,12 @@ private static bool IsPotentialReference(PredefinedType predefinedType, ISyntaxF
             : symbol.Parameters.Length;
 
         var implicitObjectKind = state.SyntaxFacts.SyntaxKinds.ImplicitObjectCreationExpression;
-        FindReferencesInDocument(state, IsRelevantDocument, CollectMatchingReferences, processResult, processResultData, cancellationToken);
-        return;
-
-        static bool IsRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
-            => syntaxTreeInfo.ContainsImplicitObjectCreation;
-
-        void CollectMatchingReferences(
-            SyntaxNode node, FindReferencesDocumentState state, Action<FinderLocation, TData> processResult, TData processResultData)
+        foreach (var newKeywordToken in newKeywordTokens)
         {
-            if (node.RawKind != implicitObjectKind)
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var node = newKeywordToken.Parent;
+            if (node is null || node.RawKind != implicitObjectKind)
                 return;
```

Comment:
>return; [](http://example.com/codeflow?start=16&length=7)

is returning still correct now that this is no longer in it's own method?

Reply:
nope!

Reply:
going to add test that demonstrates this.